grobid-core/src/main/java/org/grobid/core/data/BiblioItem.java

```diff
@@ -4182,27 +4182,20 @@ else if (bibo.getFullAuthors().size() == 1) {
                         if (StringUtils.isNotBlank(aut2.getLastName())) {
                             if (StringUtils.isNotBlank(aut.getLastName()) && aut.getLastName().equals(aut2.getLastName())) {
                                 // check also first name if present - at least for the initial
-                                if ( StringUtils.isNotBlank(aut2.getFirstName()) && StringUtils.isNotBlank(aut.getFirstName()) ) {
-                                    // we have a match (full first name)
-                                    if (StringUtils.isBlank(aut.getMiddleName()))
-                                        aut.setMiddleName(aut2.getMiddleName());
-                                    if (StringUtils.isBlank(aut.getTitle()))
-                                        aut.setTitle(aut2.getTitle());
-                                    if (StringUtils.isBlank(aut.getSuffix()))
-                                        aut.setSuffix(aut2.getSuffix());
-                                    break;
-                                } else if ( StringUtils.isNotBlank(aut.getFirstName()) && 
-                                    StringUtils.isNotBlank(aut2.getFirstName()) &&
-                                    (aut.getFirstName().length() == 1) && 
-                                    (aut.getFirstName().equals(aut2.getFirstName().substring(0,1))) ) {
-                                    // we have a match (initial)
+                                if ( StringUtils.isNotBlank(aut2.getFirstName()) && StringUtils.isNotBlank(aut.getFirstName())
+                                    &&( aut.getFirstName().equals(aut2.getFirstName()) ||
+                                    ((aut.getFirstName().length() == 1) && (aut.getFirstName().equals(aut2.getFirstName().substring(0,1)))))) {
```

Comment:
what about the initial is with a dot? so it has a length of 2? 
What about the initial is present in the `aut2` instead of `aut`? shouldnt you check that condition too?

Reply:
yes actually we should compare the initials for the firstnames  either for `aut2` or `aut` , WDYT ?

Reply:
the check (whch was like this before, so I might be wrong) first check if `aut` has length 1, and if so it compares `aut` with the first character of `aut2`

what if `aut` does not have length 1, but `aut2` is just the initial?

Reply:
yes I didn't change this check actually, it was like this..
basically if it is  not the same firstname or `aut` firstname has more than one character then it 's not considered the same author form

Reply:
what I suggest is  for firstnames to compare only the first char (initials)

Reply:
OK

I noticed another thing, at line 4205, there is `bib.setFullAuthors(bibo.getFullAuthors());` so the authors are copied anyway from the second biblio... or should I better be in bed?

Reply:
I added a couple of tests, there is an ignored one that is failing and I'm wondering what is the rule here

Reply:
yes the data about authors coming from consolidation is considered more accurate than the parsed one, 
in this PR I would like to keep the affiliations if none fetched from the consolidation which seems to be more frequent

Reply:
regarding the test, so far the affiliations from crossref are just raw and not deserialised, so maybe we could keep the affiliations parsed from the pdf ? I'm not sure how good are the affiliations from crossref actually